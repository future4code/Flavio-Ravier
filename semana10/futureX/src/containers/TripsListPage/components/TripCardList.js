import React, { Component } from "react";
import { connect } from "react-redux";
import { push } from "connected-react-router";
import {
        Button,
        Paper,}
                from "@material-ui/core/";
import styled from "styled-components";
import {routes} from "../../Router"
import { Grid } from 'material-ui'
import {Planet} from '../../../img/index'
import Typography from '@material-ui/core/Typography'

import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';

const Hh = styled.h1`
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
    text-align:center;
`;


//mock de dados
const allTrips =
  
    {plutao: {
          "id": "KmwhN0tLvE6MQM5WxmOY",
          "name": "Picnic de Inverno em Plutão",
          "description": "Único tour que fazemos em planeta anão no sistema solar! Levem casacos que a previsão é de −230 °C",
          "planet": "Plutão",
          "durationInDays": 980,
          "date": "21/12/20",
          "img":Planet.plutao,
     } ,
     marte: {
          "id": "Qi4MEg5TcleOdNWqYmNb",
          "name": "Festança Marciana",
          "description": "Uma viagem bem legal, na melhor época de marte",
          "planet": "Marte",
          "durationInDays": 228,
          "date": "21/12/21",
          "img":Planet.marte,
     },
     jupiter: {
          "id": "Rozpqj50h8inVPKXVbPz",
          "name": "Multi luau em Jupiter",
          "description": "Noite mágica, com vista para as 69 luas de Jupiter",
          "planet": "Jupiter",
          "durationInDays": 540,
          "date": "21/12/20",
          "img":Planet.jupiter,
          "aprovado":""
     },
      neturno: {
          "id": "aO2T6Bif7zDisYd8s7dU",
          "planet": "Netuno",
          "durationInDays": 540,
          "date": "21/12/20",
          "name": "Surfando em Netuno",
          "description": "Nenhum surfista intergalático pode ficar fora dessa!",
          "img":Planet.neturno,
      }}
    



function TripCard(props){
  return (
    <Card  style={{ marginLeft:30, marginTop:10, marginBottom:10, width:275, height:375}}>
        <CardActionArea>
            <CardMedia 
                height="160"
                component="img"
                alt="Planeta de Destino"
                image={props.img}
                title={props.planet}>
            </CardMedia>
            <Paper elevation={5} style={{ marginLeft:15, marginTop:5, marginBottom:5, width:150, height:20}}>
                <Typography variant="body2" color="textSecondary" component="p" align='center'>{props.durationInDays} Dias de Trip!</Typography>
            </Paper>  
            <CardContent style={{padding:0,paddingLeft:7}}>
                <Typography gutterBottom variant="h5" component="h2"> 
                  {props.name}
                </Typography>
                <Typography variant="body2" color="textSecondary" component="p">
                  {props.description}
                </Typography>
                <CardActions style={{alignItems:"center",justifyContent:"center"}}>
                      <Button size="small" color="secondary">
                        Inscreva-se Já!
                      </Button>
                </CardActions>
            </CardContent> 
        </CardActionArea>
    </Card>
);
}

    
  

function mapToDispatch(dispatch){
  return {
    goToApplicationForm: () => dispatch(push(routes.application_form))
  }
}

export default connect(
  null,
  mapToDispatch)(TripCard);

/*   class ApplicationPage extends Component {
    constructor(props) {
      super(props);
      };
      componentDidMount(){
        debugger
        console.log(this.props)
      }
    render() {
      return (
          <Card  style={{ marginLeft:30, marginTop:10, marginBottom:10, width:275, height:375}}>
              <CardActionArea>
                  <CardMedia 
                      height="160"
                      component="img"
                      alt="Planeta de Destino"
                      image={this.props.img}
                      title={this.props.planet}>
                  </CardMedia>
                  <Paper elevation={5} style={{ marginLeft:15, marginTop:5, marginBottom:5, width:150, height:20}}>
                      <Typography variant="body2" color="textSecondary" component="p" align='center'>{this.props.durationInDays} Dias de Trip!</Typography>
                  </Paper>  
                  <CardContent style={{padding:0,paddingLeft:7}}>
                      <Typography gutterBottom variant="h5" component="h2"> 
                        {this.props.name}
                      </Typography>
                      <Typography variant="body2" color="textSecondary" component="p">
                        {this.props.description}
                      </Typography>
                      <CardActions style={{alignItems:"center",justifyContent:"center"}}>
                            <Button size="small" color="secondary">
                              Inscreva-se Já!
                            </Button>
                      </CardActions>
                  </CardContent> 
              </CardActionArea>
          </Card>
      );
    }
  }
  
  function mapToDispatch(dispatch){
    return {
      goToApplicationForm: () => dispatch(push(routes.application_form))
    }
  }
  
  export default connect(
    null,
    mapToDispatch)(ApplicationPage);
   */