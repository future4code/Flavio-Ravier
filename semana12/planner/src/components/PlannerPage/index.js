import React from "react";
import styled from "styled-components";
import Post from "./../../containers/Post/index"
import Planner from "./../../containers/Planner"

const PlannerWrapper = styled.div`
  width: 100%;
  height: 70vh;
  gap: 30px;
  place-content: top;
  justify-items: center;
  align-items:center;
  display: flex;
  flex-direction: column;
  margin-top:30px;
`;


function PlannerPage() {
	
    	return (
    		<PlannerWrapper>
				<Post></Post>
				<Planner></Planner>
    		</PlannerWrapper>
    	)
  	
}

export default PlannerPage;
