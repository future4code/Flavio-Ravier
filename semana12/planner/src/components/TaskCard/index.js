import React, { Component } from "react";
import styled from "styled-components";
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';

const TaskCardWrapper = styled(Paper)`
  width: 130px;
  height: 80px;
  justify-content: space-evenly;
  justify-items: center;
  align-items:top;
  text-align:center;
  display: flex;
  flex-direction:column;
  margin-top:10px;
  padding:5px;
`;

export class TaskCard extends Component {

    render() {
        return (
            <TaskCardWrapper>
                <Typography variant="overline" color="primary" component="h5">
                    {this.props.taskText}
                   
                </Typography>
            </TaskCardWrapper>
        )
    }
}
export default (TaskCard)