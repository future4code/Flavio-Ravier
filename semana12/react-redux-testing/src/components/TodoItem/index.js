import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const TodoListItemWrapper = styled.div`
  display: flex;
  width: 300px;
  justify-content: space-between;
`;

export const TodoListItem = styled.li`
  list-style: decimal;
  margin: 5px 0;
  text-decoration: ${props => (props.completed ? "line-through" : "unset")};
`;

export const DeleteButton = styled.span`
  color: blue;
  cursor: pointer;

  :hover {
    font-weight: 700;
  }
`;

export const TodoItem = props => {
  const handleOnClick = () => {
    props.onCompleteTodo(props.todo.id);
  };

  const handleOnDelete = () => {
    props.onDeleteTodo(props.todo.id);
  };

  return (
    <TodoListItemWrapper>
      <TodoListItem completed={props.todo.completed} onClick={handleOnClick}>
        {props.todo.text}
      </TodoListItem>
      <DeleteButton onClick={handleOnDelete}>X</DeleteButton>
    </TodoListItemWrapper>
  );
};

TodoItem.propTypes = {
  todo: PropTypes.any.isRequired,
  onCompleteTodo: PropTypes.func.isRequired,
  onDeleteTodo: PropTypes.func.isRequired
};
