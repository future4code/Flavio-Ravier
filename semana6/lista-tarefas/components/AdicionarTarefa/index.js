import React from 'react';
import styled from 'styled-components';
import Proptypes from 'prop-types'