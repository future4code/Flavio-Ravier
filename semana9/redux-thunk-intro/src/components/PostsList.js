import React from "react";
import { connect } from 'react-redux';
import PostCard from "./PostCard/PostCard";
import { getPosts } from '../actions/post'

class PostsList extends React.Component {

  componentDidMount(){
    this.props.getAllPosts();
  }

  render() {
    return this.props.postsList.map(post => (<PostCard post={post} />)
    );
  }
}

const mapStateToProps = (state) => ({
  postsList: state.posts.allPosts,
});

const mapDispatchToProps = (dispatch) => ({
  getAllPosts: () => dispatch(getPosts()),
});

export default connect(mapStateToProps, mapDispatchToProps)(PostsList)
