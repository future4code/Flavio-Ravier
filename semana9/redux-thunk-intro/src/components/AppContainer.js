import React, { Component } from "react";
import PostForm from "./PostForm/PostForm";
import PostsList from "./PostsList";
import styled from "styled-components";

const Container = styled.div`
  display: grid;
  grid-template-columns: 400px;
  justify-content: center;
  gap: 15px;
`;

class AppContainer extends Component {
  render() {
    return (
      <Container>
        <PostForm />
        <PostsList />
      </Container>
    );
  }
}

export default AppContainer;
