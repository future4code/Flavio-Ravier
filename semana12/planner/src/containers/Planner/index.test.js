import React from "react";
import {TaskCard} from "./../../components/TaskCard";
import {Planner} from './index';
import {shallow} from 'enzyme';
import renderer from "react-test-renderer";



describe("Test Planner", () => {
    test("Swallow Planner", () => {
        const listMockMon = [
            {id:1, text:"Bal", day:"monday"},
            {id:2, text:"B2al", day:"monday"}
        ]

        const listMockFri = [
            {id:1, text:"Bal", day:"friday"},
            {id:2, text:"B2wqal", day:"friday"},
            {id:3, text:"B2dqdwqal", day:"friday"}
        ] 
             
        

        const component = shallow(<Planner getTaskList={jest.fn} listOfMon={listMockMon} listOfFri={listMockFri} 
            listOfTue={[]} listOfWed={[]} listOfThu={[]}
            listOfSat={[]} listOfSun={[]}/>)
        
        const taskCards = component.find(TaskCard)
        
        //ele le que existem 5 taskcards
        expect(taskCards).toHaveLength(5)
     })
     
    

    test('Snapshot', () => {
        const tree = renderer.create(<Planner getTaskList={jest.fn} listOfMon={[]} listOfFri={[]} 
            listOfTue={[]} listOfWed={[]} listOfThu={[]} listOfThu={[]} 
            listOfSat={[]} listOfSun={[]}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
}) 
