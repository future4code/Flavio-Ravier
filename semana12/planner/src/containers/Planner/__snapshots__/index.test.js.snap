// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Planner Snapshot 1`] = `
<div
  className="sc-bwzfXH fIVSIO"
>
  <div
    className="sc-bxivhb dzIoij"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary"
      style={
        Object {
          "margin": "0 0 15px 0",
        }
      }
    >
      Segunda
    </h5>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-vertical"
  />
  <div
    className="sc-bxivhb dzIoij"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary"
      style={
        Object {
          "margin": "0 0 15px 0",
        }
      }
    >
      Terça
    </h5>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-vertical"
  />
  <div
    className="sc-bxivhb dzIoij"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary"
      style={
        Object {
          "margin": "0 0 15px 0",
        }
      }
    >
      Quarta
    </h5>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-vertical"
  />
  <div
    className="sc-bxivhb dzIoij"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary"
      style={
        Object {
          "margin": "0 0 15px 0",
        }
      }
    >
      Quinta
    </h5>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-vertical"
  />
  <div
    className="sc-bxivhb dzIoij"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary"
      style={
        Object {
          "margin": "0 0 15px 0",
        }
      }
    >
      Sexta
    </h5>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-vertical"
  />
  <div
    className="sc-bxivhb dzIoij"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary"
      style={
        Object {
          "margin": "0 0 15px 0",
        }
      }
    >
      Sábado
    </h5>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-vertical"
  />
  <div
    className="sc-bxivhb dzIoij"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 MuiTypography-colorSecondary"
      style={
        Object {
          "margin": "0 0 15px 0",
        }
      }
    >
      Domingo
    </h5>
  </div>
</div>
`;
